test/e2e/delete_test.go

```diff
@@ -42,39 +34,33 @@ func testWindowsNodeDeletion(t *testing.T) {
 			windowsMachineSetWithLabel = machineSet.Name
 		}
 	}
-	windowsMachineSet := &mapi.MachineSet{}
-	err = framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: windowsMachineSetWithLabel,
-		Namespace: "openshift-machine-api"}, windowsMachineSet)
-	require.NoError(t, err)
+	windowsMachineSet, err := testCtx.client.Machine.MachineSets(clusterinfo.MachineAPINamespace).Get(context.TODO(),
+		windowsMachineSetWithLabel, meta.GetOptions{})
+	require.NoError(t, err, "error getting Windows MachineSet")
 	// Reset the number of nodes to be deleted to 0
 	gc.numberOfNodes = 0
 	// Delete the Windows VM that got created.
 	windowsMachineSet.Spec.Replicas = &gc.numberOfNodes
-	if err := framework.Global.Client.Update(context.TODO(), windowsMachineSet); err != nil {
-		t.Fatalf("error updating windowsMachineSet custom resource  %v", err)
-	}
+	_, err = testCtx.client.Machine.MachineSets(clusterinfo.MachineAPINamespace).Update(context.TODO(), windowsMachineSet,
+		meta.UpdateOptions{})
+	require.NoError(t, err, "error updating Windows MachineSet")
 	// we are waiting 10 minutes for all windows machines to get deleted.
 	err = testCtx.waitForWindowsNodes(gc.numberOfNodes, true, true, false)
-	if err != nil {
-		t.Fatalf("windows node deletion failed  with %v", err)
-	}
+	require.NoError(t, err, "Windows node deletion failed")
 
 	// Cleanup all the MachineSets created by us.
-	for _, machineSetName := range e2eMachineSetNames {
-		machineSet := &mapi.MachineSet{}
-		err = framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: machineSetName,
-			Namespace: "openshift-machine-api"}, machineSet)
-		assert.NoError(t, framework.Global.Client.Delete(context.TODO(), machineSet))
+	for _, machineSet := range machineSetList.Items {
+		testCtx.deleteMachineSet(&machineSet)
```

Comment:
don't we need to assert for error here?

Reply:
We should not fail the test because clean up failed. I suggest we log the error here.

Reply:
Though such behavior change should not be done in this PR. This PR should be mechanical with no change in behavior or policy unless required for the SDK 1.0 migration.

Reply:
I misread your previous suggestion @aravindhp https://github.com/openshift/windows-machine-config-operator/pull/314#discussion_r579431579
Thought you were saying that we shouldnt capture the error. I'll correct it